src/pages/Assets/Asset.tsx

```diff
@@ -13,30 +14,36 @@ export interface MatchParams {
 }
 
 export const Asset = () => {
-  const [asset, setAsset] = useState<AssetMarketData>()
-  const [loading, setLoading] = useState<boolean>(false)
+  const [isLoaded, setIsLoaded] = useStateIfMounted<boolean>(false)
+  const [asset, setAsset] = useStateIfMounted<AssetMarketData | undefined>(undefined)
+
   let { network, address } = useParams<MatchParams>()
   const getAssetData = useGetAssetData()
   const getPrice = useCallback(async () => {
-    setLoading(true)
     const asset = await getAssetData({
       chain: ChainTypes.ETH,
       network: NetworkTypes.MAINNET,
       tokenId: address
     })
     if (asset) setAsset(asset)
-    setLoading(false)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [address, getAssetData])
 
   useEffect(() => {
-    getPrice()
+    ;(async () => {
+      setIsLoaded(false)
+      setTimeout(async () => {
+        await getPrice()
+        setIsLoaded(true)
+      }, 750)
+    })()
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [network, address])
 
   return (
-    <Page style={{ flex: 1 }} loading={loading} error={!asset}>
+    <Page style={{ flex: 1 }} key={address}>
       <Flex role='main' flex={1} height='100%'>
-        {asset && <AssetDetails asset={asset} />}
+        <AssetDetails asset={asset ?? ({} as AssetMarketData)} isLoaded={isLoaded} />
```

Comment:
why are we casting here?

Reply:
Its expecting an AssetMarketData object, and we want to be able to display skeletons without an actual asset loaded yet

Reply:
this also allows us to pass garbage data when it's not loading though right?

Reply:
i'd suggest nullish coalescing to a valid dummy object rather than casting and potentially creating a type error here fwiw